### PR TITLE
fix(media): introduce delete multiple files endpoint

### PR DESCRIPTION
**Pull request by dcshzj**
> [!NOTE]
> Large PR due to unit tests, feel free to browse by commits

## Problem

The current delete file endpoint often encounters the repo lock, prevent users from deleting multiple files at once.

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible with ALL of the following feature flags in this [doc](https://www.notion.so/opengov/Existing-feature-flags-518ad2cdc325420893a105e88c432be5)

**Improvements**:

- A new endpoint has been introduced to allow for deleting of multiple files in a single repo lock transaction.

## Tests

- Unit tests.
- Smoke tests (written on https://github.com/isomerpages/isomercms-frontend/pull/1753)

## Deploy Notes

*None*

### CODE REVIEW

File: src/services/db/GitFileSystemService.ts
Comment (harishv7) [inline src/services/db/GitFileSystemService.ts:652]: why can't we do a `git add .`?

File: src/services/db/GitFileSystemService.ts
Comment (harishv7) [inline src/services/db/GitFileSystemService.ts:1314]: would this mean a single failure in a batch of 20 deletions will rollback all deletions? we don't support partial success of the request?

File: src/services/db/GitFileCommitService.ts
Comment (harishv7) [inline src/services/db/GitFileCommitService.ts:277]: staging lite doesn't contain images or files right? in that case why do we need to update stg lite branch?

File: src/services/db/GitFileSystemService.ts
Comment (dcshzj) [reply src/services/db/GitFileSystemService.ts:652]: We don't have sufficient guarantees that doing a `git add .` only adds the files that were indeed changed by the user. There may be other changes that were made on the repo (which may be there for unknown reasons) but not yet committed, we don't want to accidentally add them here.

File: src/services/db/GitFileSystemService.ts
Comment (dcshzj) [reply src/services/db/GitFileSystemService.ts:1314]: This is more for simplicity. There can be many reasons why failure happens, so I decided to treat it as an atomic transaction instead of having some success some failure. On the frontend, we don't also have an easy way to show the user which files failed to delete, so I decided to go for getting the user to retry again.

File: src/services/db/GitFileCommitService.ts
Comment (dcshzj) [reply src/services/db/GitFileCommitService.ts:277]: We don't have that guarantee at this stage, because we only know the `filePath` and `sha`. Doing this is safer in case we somehow reuse this function to bulk delete pages in the future.

File: src/services/db/RepoService.ts
Comment (kishore03109) [inline src/services/db/RepoService.ts:246]: perf: we can reduce a redundant `super.getRepoInfo(sessionData)` in this control flow by adding this line before `const { private: isPrivate } = await super.getRepoInfo(sessionData)`

File: src/services/fileServices/MdPageServices/__tests__/MediaFileService.spec.js
Comment (kishore03109) [inline src/services/fileServices/MdPageServices/__tests__/MediaFileService.spec.js:246]: was the point of this to check for encoding issues hahah

File: src/services/db/GitFileSystemService.ts
Comment (kishore03109) [inline src/services/db/GitFileSystemService.ts:1264]: nit: no need casing already since we have already declared in the func args

**Review by kishore03109 — APPROVED** [review]
mostly nits, please dismiss my review if my comment regarding the line `return result.isOk() && result.value` has a wrong assumption

File: src/services/db/GitFileSystemService.ts
Comment (dcshzj) [reply src/services/db/GitFileSystemService.ts:1300]: Hmm yes, the `result.value` is the one that does the check for the unwrapped `ok`. Effectively there are 3 cases:
1. If an error occurred, `result.isOk()` is false, so the entire return is false.
2. If no error occurred but there is a sha mismatch, then result.value is false, so true && false becomes false.
3. If no error occurred and everything is good, then result.value is true, so true && true becomes true, which is what we want.

This is only specific for a single item, so it will loop through all the items before we determine the value of `isFileConflict`, which will then determine if we continue the function execution or not.

File: src/services/db/GitFileSystemService.ts
Comment (dcshzj) [reply src/services/db/GitFileSystemService.ts]: Oops thanks! Removed in ad2521fd.

File: src/services/db/RepoService.ts
Comment (dcshzj) [reply src/services/db/RepoService.ts:245]: Oh good idea, updated in ad2521fd.

File: src/services/db/GitFileSystemService.ts
Comment (dcshzj) [reply src/services/db/GitFileSystemService.ts]: Updated in ad2521fd!

File: src/services/fileServices/MdPageServices/__tests__/MediaFileService.spec.js
Comment (dcshzj) [reply src/services/fileServices/MdPageServices/__tests__/MediaFileService.spec.js:246]: Oops not really haha, it was just a randomly chosen special character

File: src/services/db/GitFileSystemService.ts
Comment (dcshzj) [reply src/services/db/GitFileSystemService.ts:1324]: Hmm no good arguments for this other than just being cautious about it, basically not doing a force delete unless absolutely necessary (for directories), so we avoid masking potential errors.

Comment (kishore03109) [reply src/services/db/GitFileSystemService.ts:1352]: I think the [-f flag ignores non-existent files and doesn't ask to prompt](https://man7.org/linux/man-pages/man1/rm.1.html), seems like safer here?

but is a nit so i dont feel strong about this too, let you make the final decision

Comment (dcshzj) [reply src/services/db/GitFileSystemService.ts:1352]: I was thinking it ignores non-existent files but if at this stage we are deleting a non-existent file then we might have encountered some issue where someone else deleted the file, which might be grounds for a conflict :(

**Pull request merged**
